src/main/java/fr/sncf/osrd/railml/RMLSwitchIL.java

```diff
@@ -0,0 +1,33 @@
+package fr.sncf.osrd.railml;
+
+import fr.sncf.osrd.infra.InvalidInfraException;
+import fr.sncf.osrd.railjson.schema.infra.RJSSwitch;
+import org.dom4j.Document;
+import org.dom4j.Element;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+// https://wiki3.railml.org/index.php?title=IL:switchIL
+public class RMLSwitchIL {
+
+    static ArrayList<RJSSwitch> parse(
+            Document document,
+            HashMap<String, RMLSwitchIS> switchesIS
+    ) throws InvalidInfraException {
+        var xpath = "/railML/interlocking/assetsForIL/switchesIL/switchIL";
+        var res = new ArrayList<RJSSwitch>();
+        for (var switchNode : document.selectNodes(xpath)) {
+            var switchIL = (Element) switchNode;
+            var throwTimeIso8601 = switchIL.attributeValue("typicalThrowTime", "PT0S");
+            double throwTime = java.time.Duration.parse(throwTimeIso8601).toMillis();
+            double throwTimeSeconds = throwTime / 1000.;
+            var id = switchIL.attributeValue("id");
+            var switchIS = switchesIS.get(id);
```

Comment:
`switchIL.id` shouldn't match `switchIS.id`. Use the `refersTo` component instead.